### PR TITLE
switch to use region specific endpoints to compute correct v4 signature

### LINKED ISSUE #133: SignatureDoesNotMatch
Hi!
I have got a problem using the latest version of s3fs.
OS: CentOS Linux release 7.0.1406 (Core)
Kernel: 3.14.18-x86_64-jb1 #2 SMP Mon Sep 15 05:23:31 CEST 2014 x86_64 x86_64 x86_64 GNU/Linux
S3 Bucket Region: eu-central-1
Mountcommand: s3fs -d mybucketname /s3mnt -oendpoint=eu-central-1 -oallow_other

Every time i try to copy a file lager than 5MB to the S3 bucket, i get this error:
SignatureDoesNotMatch: The request signature we calculated does not match the signature you provided. Check your key and signing method.

Copying files up to ~5MB to the bucket works correctly. And receiving files from amazon also.

I hope someone could help me.

### PR DESCRIPTION
**Pull request by kahing**
fix #133

### CODE REVIEW

File: src/s3fs.cpp [hunk 1]
```diff
@@ -3102,6 +3102,9 @@ static int s3fs_check_service(void)
       LOWSYSLOGPRINT(LOG_ERR, "Could not connect wrong region %s, so retry to connect region %s.", endpoint.c_str(), expectregion.c_str());
       FPRN("Could not connect wrong region %s, so retry to connect region %s.", endpoint.c_str(), expectregion.c_str());
       endpoint = expectregion;
+      if (S3fsCurl::IsSignatureV4() && host == "http://s3.amazonaws.com") {
```
Comment (andrewgaul) [inline src/s3fs.cpp]: Rewrite https:// as well?

File: src/s3fs.cpp [hunk 2]
```diff
@@ -3102,6 +3102,13 @@ static int s3fs_check_service(void)
       LOWSYSLOGPRINT(LOG_ERR, "Could not connect wrong region %s, so retry to connect region %s.", endpoint.c_str(), expectregion.c_str());
       FPRN("Could not connect wrong region %s, so retry to connect region %s.", endpoint.c_str(), expectregion.c_str());
       endpoint = expectregion;
+      if (S3fsCurl::IsSignatureV4()) {
+          if (host == "http://s3.amazonaws.com") {
+              host = "http://s3-" + endpoint + ".amazonaws.com";
+          } else if (host == "http://s3.amazonaws.com") {
```
Comment (andrewgaul) [inline src/s3fs.cpp]: Typo, should be https.

**Comment by andrewgaul** [comment]
:+1:

**Comment by ggtakec** [comment]
@kahing and @andrewgaul 
Thanks.

**Pull request merged**